azurerm/internal/services/network/network_connection_monitor_resource.go

```diff
@@ -200,10 +226,50 @@ func resourceNetworkConnectionMonitor() *schema.Resource {
 							},
 						},
 
+						"included_addresses": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+								ValidateFunc: validation.Any(
+									validation.IsIPv4Address,
+									validation.IsIPv6Address,
+									validation.IsCIDR,
+								),
+							},
+						},
+
+						"resource_id": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+							ValidateFunc: validation.Any(
+								computeValidate.VirtualMachineID,
+								logAnalyticsValidate.LogAnalyticsWorkspaceID,
+								networkValidate.SubnetID,
+								networkValidate.VirtualNetworkID,
+							),
+						},
+
+						"type": {
```

Comment:
is this the type of resource ID? could we not figure it out from the resource id format/fields?

Reply:
No. It's network connection monitor endpoint type. Not all endpoint types are expressed by resource id. So we cannot figure it out from the resource id.

Reply:
👍